packages/modules/e3dc/config.py

```diff
@@ -0,0 +1,71 @@
+from modules.common.component_setup import ComponentSetup
+from helpermodules.auto_str import auto_str
+
+
+@auto_str
+class E3dcConfiguration:
+    def __init__(self, address: str = None):
+        self.address = address
+
+
+@auto_str
+class E3dc:
+    def __init__(self,
+                 name: str = "e3dc",
+                 type: str = "e3dc",
+                 id: int = 0,
+                 configuration: E3dcConfiguration = None) -> None:
+        self.name = name
+        self.type = type
+        self.id = id
+        self.configuration = configuration or E3dcConfiguration()
+
+
+@auto_str
+class E3dcBatConfiguration:
+    def __init__(self):
+        pass
+
+
+@auto_str
+class E3dcBatSetup(ComponentSetup[E3dcBatConfiguration]):
+    def __init__(self,
+                 name: str = "e3dc Speicher",
+                 type: str = "bat",
+                 id: int = 0,
+                 configuration: E3dcBatConfiguration = None) -> None:
+        super().__init__(name, type, id, configuration
+                         or E3dcBatConfiguration())
+
+
+@auto_str
+class E3dcCounterConfiguration:
+    def __init__(self):
+        pass
+
+
+@auto_str
+class E3dcCounterSetup(ComponentSetup[E3dcCounterConfiguration]):
+    def __init__(self,
+                 name: str = "e3dc Zähler",
+                 type: str = "counter",
+                 id: int = 0,
+                 configuration: E3dcCounterConfiguration = None) -> None:
+        super().__init__(name, type, id, configuration or
+                         E3dcCounterConfiguration())
+
+
+@auto_str
+class E3dcInverterConfiguration:
+    def __init__(self, read_ext: int = 0):
```

Comment:
Ist zwar zu spät, aber die Anmerkungen bitte beim PR #2496 berücksichtigen.
Der Datentyp `read_ext: int` passt hier nicht wirklich, da nur 0 oder 1 gültige Werte sind. Besser ein bool verwenden. Die Kompatibilität zu 1.9 mit 0/1 wird in der read_legacy_bat hergestellt. Z.B. `read_external = read_ext == 1`.